specification.markdown

```diff
@@ -97,242 +170,298 @@ followed by the Data Size, and then the non-interpreted Binary itself:
-   <td>Contains elements that will be used to compute the signature.</td>
-  </tr>
-  <tr class="version2">
-   <td>SignatureElementList<a name="SignatureElementList"></a></td>
-   <td>3+</td>
-   <td>[7E][7B]</td>
-   <td>-</td>
-   <td>*</td>
-   <td>-</td>
-   <td>-</td>
-   <td>sub-elements</td>
-   <td>A list consists of a number of consecutive elements that represent one
-    case where data is used in signature. Ex: <i>Cluster|Block|BlockAdditional</i>
-    means that the BlockAdditional of all Blocks in all Clusters is used for
-    encryption.</td>
-  </tr>
-  <tr class="version2">
-   <td>SignedElement<a name="SignedElement"></a></td>
-   <td>4+</td>
-   <td>[65][32]</td>
-   <td>-</td>
-   <td>*</td>
-   <td>-</td>
-   <td>-</td>
-   <td>binary</td>
-   <td>An element ID whose data will be used to compute the signature.</td>
-  </tr>
- </tbody>
-</table>
+## Structure
+
+Syntactically an EBML document is a list of EBML elements. Each element
+has three parts; an Element ID, an Element Data Size and the element
+data payload. The element data payload is then either data, implicitly
+typed by the Element ID, or a list of EBML elements.
+
+    EBML       = *ELEMENT
+    ELEMENT    = ELEMENT_ID SIZE DATA
+    DATA       = VALUE / *ELEMENT
+    ELEMENT_ID = VINT
+    SIZE       = VINT
+
+EBML uses big endian/network order byte order, i.e. most significant
+bit first. All of the tokens above are byte aligned.
+
+## Variable Size Integer
+
+For both Element ID and Element Data Size EBML uses a variable size
+integer, coded according to a schema similar to that of UTF-8 encoding.
+The variable size integer begins with zero or more zero bits to define
+the width of the integer. Zero zeroes means a width of one byte, one
+zero a width of two bytes etc. The zeroes are followed by a marker of
+one set bit and then follows the actual integer value. The integer data
+consists of alignment data and tail data. The alignment data together
+with the width descriptor and the marker makes up one or more complete
+bytes. The tail data is as many bytes as there were zeroes in the width
+descriptor, i.e. width-1.
+
+    VINT           = VINT_WIDTH VINT_MARKER VINT_DATA
+    VINT_WIDTH     = *%b0
+    VINT_MARKER    = %b1
+    VINT_DATA      = VINT_ALIGNMENT VINT_TAIL
+    VINT_ALIGNMENT = *BIT
+    VINT_TAIL      = *BYTE
+
+An alternate way of expressing this is the following definition, where
+the width is the number of levels of expansion.
+
+VINT           = ( %b0 VINT 7BIT ) / ( %b1 7BIT )
+
+Some examples of the encoding of integers of width 1 to 4. The x:es
+represent bits where the actual integer value would be stored.
+
+Width | Size | Representation
+------|------|----------------------------------------
+1     | 2^7  | 1xxx xxxx
+2     | 2^14 | 01xx xxxx xxxx xxxx
+3     | 2^21 | 001x xxxx xxxx xxxx xxxx xxxx
+4     | 2^28 | 0001 xxxx xxxx xxxx xxxx xxxx xxxx xxxx
+
+## Elements semantic
+
+### Element Template
+
+Element Name:
+
+    Level:
```

Comment:
Explain that a + means it's nested. It may also be another line:
`Nested: Yes/No`